crates/red_knot_python_semantic/src/types/narrow.rs

```diff
@@ -385,10 +385,11 @@ impl<'db> NarrowingConstraintsBuilder<'db> {
         let scope = self.scope();
         let inference = infer_expression_types(self.db, expression);
 
+        let expr_ty = inference.expression_ty(expr_call.func.scoped_expression_id(self.db, scope));
```

Comment:
This is neither the type of `expr_call`, nor the type of `expression`, so the name is slightly confusing. Maybe rename it to
```suggestion
        let callable_ty = inference.expression_ty(expr_call.func.scoped_expression_id(self.db, scope));
```